### PR TITLE
feat: add TO_BYTES/FROM_BYTES functions for bytes/string conversions

### PR DESCRIPTION
**Pull request by spena**
### Description 
Adds `TO_BYTES` and `FROM_BYTES` functions. Supported encodings are `hex`, `ascii`, `utf8`, and `base64`.

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/ksqldb-reference/scalar-functions.md [hunk 1]
```diff
@@ -729,6 +729,17 @@ multiple elements, like those containing wildcards, aren't supported.
...
+### `FROM_BYTES`
+
+Since: - 0.21
+
+```sql
+FROM_BYTES(bytes, encoding)
+```
+
+Converts a BYTES column to a STRING in the specified encoding type.
+Supported encoding types are: 'hex', 'utf8', 'ascii', 'base64'.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/scalar-functions.md:741]:
```suggestion
Supported encoding types are: `ex`, `utf8`, `ascii`, and `base64`.
```

File: docs/developer-guide/ksqldb-reference/scalar-functions.md [hunk 2]
```diff
@@ -1046,6 +1057,17 @@ the string.
...
+### `TO_BYTES`
+
+Since: - 0.21
+
+```sql
+TO_BYTES(string, encoding)
+```
+
+Converts a STRING column in the specified encoding type to a BYTES column.
+Supported encoding types are: 'hex', 'utf8', 'ascii', 'base64'.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/scalar-functions.md:1069]:
```suggestion
Supported encoding types are: `hex`, `utf8`, `ascii`, `base64`.
```

**Review by JimGalasyn — COMMENTED** [review]
LGTM, with a couple of suggestions.

File: docs/developer-guide/ksqldb-reference/scalar-functions.md (hunk 2 above)
Comment (spena) [reply docs/developer-guide/ksqldb-reference/scalar-functions.md]: Done

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/FromBytes.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    category = FunctionCategory.STRING,
+    description = "Converts a BYTES value to STRING in the specified encoding. "
+        + "The accepted encoders are 'hex', 'utf8', 'ascii', and 'base64'.",
+    author = KsqlConstants.CONFLUENT_AUTHOR
+)
+public class FromBytes {
+  @Udf(description = "Converts a BYTES value to STRING in the specified encoding. "
+      + "The accepted encoders are 'hex', 'utf8', 'ascii', and 'base64'.")
+  public String fromBytes(final ByteBuffer value, final String encoding) {
+    return (value == null) ? null : BytesUtils.encode(value.array(), encoding);
```
Comment (jzaralim) [inline ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/FromBytes.java:36]: The Protobuf deserializer deserializes to a read-only byte buffer, so `value.array()` might throw.

File: ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/string/FromBytesTest.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+import java.nio.ByteBuffer;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThrows;
+
+public class FromBytesTest {
```
Comment (jzaralim) [inline ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/string/FromBytesTest.java:28]: Can we also have a test for an empty byte array?

File: ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/string/ToBytesTest.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThrows;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+
+public class ToBytesTest {
```
Comment (jzaralim) [inline ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/string/ToBytesTest.java:28]: Can we also have a test for an empty string?

File: ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/string/FromBytesTest.java (hunk 4 above)
Comment (spena) [reply ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/string/FromBytesTest.java:28]: Done

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/FromBytes.java (hunk 3 above)
Comment (spena) [reply ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/FromBytes.java]: Done

File: ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/planned/PlannedTestGeneratorTest.java [hunk 6]
```diff
@@ -35,7 +35,6 @@
    * with your change. Otherwise, {@link PlannedTestsUpToDateTest} fill fail if there are missing
    * or changed query plans.
    */
-  @Ignore("Comment me out to regenerate the historic plans")
```
[the remark this answers is not in the document]
Comment (spena) [reply ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/planned/PlannedTestGeneratorTest.java]: Done

**Review by jzaralim — APPROVED** [review]
LGTM
